api_app/analyzers_manager/file_analyzers/yara_scan.py

```diff
@@ -69,75 +62,172 @@ def _validated_matches(self, rules: yara.Rules) -> list:
+        response = requests.get(url, stream=True)
+        try:
+            response.raise_for_status()
+        except Exception as e:
+            logger.exception(e)
+        else:
+            zipfile_ = zipfile.ZipFile(io.BytesIO(response.content))
+            zipfile_.extractall(directory)
+
+    @classmethod
+    def _update_repository(cls, url: str):
+        if url.endswith(".zip"):
+            cls._download_or_update_zip_repository(url)
+        else:
+            cls._download_or_update_git_repository(url)
+
+    @classmethod
+    def update_rules(cls):
+        logger.info("Starting updating yara rules")
         analyzer_config = AnalyzerConfig.all()
-        found_yara_dirs = []
+        urls = set()
         for analyzer_name, ac in analyzer_config.items():
-            if analyzer_name.startswith("Yara_Scan") and ac.param_values.get(
-                "update", True
+            if (
+                ac.python_module == f"{cls.__module__.split('.')[-1]}.{cls.__name__}"
+                and ac.disabled is False
             ):
-                logger.info(f"working on analyzer {analyzer_name}")
-                yara_dirs = ac.param_values.get("git_repo_main_dir", [])
-                if not yara_dirs:
-                    yara_dirs = ac.param_values.get("directories_with_rules", [])
-                yara_urls = ac.param_values.get("url", [])
-                if yara_urls:
-                    for yara_url, yara_dir in zip(yara_urls, yara_dirs):
-                        response = requests.get(yara_url, stream=True)
-                        zipfile_ = zipfile.ZipFile(io.BytesIO(response.content))
-                        zipfile_.extractall(yara_dir)
-                        logger.info(f"download {yara_url}")
-                else:
-                    for yara_dir in yara_dirs:
-                        if os.path.isdir(yara_dir):
-                            repo = Repo(yara_dir)
-                            o = repo.remotes.origin
-                            git = repo.git
-                            git.config("--global", "--add", "safe.directory", yara_dir)
-                            o.pull()
-                            logger.info(f"pulled repo on {yara_dir} dir")
-                        else:
-                            logger.warning(f"yara dir {yara_dir} does not exist")
-                found_yara_dirs.extend(yara_dirs)
-        return found_yara_dirs
+                new_urls = ac.param_values.get("repositories", [])
+                logger.info(f"Adding urls {new_urls}")
+                urls.update(new_urls)
+
+                # we are downloading even custom signatures for each analyzer
+                for plugin in PluginConfig.objects.filter(
+                    plugin_name=analyzer_name,
+                    type=PluginConfig.PluginType.ANALYZER,
+                    config_type=PluginConfig.ConfigType.PARAMETER,
+                    attribute="repositories",
```

Comment:
shouldn't this be filtered by organization/user?

Reply:
The `updated` is generic and not linked to a single user/org by definition.

Reply:
There is no permission problem here because the analyzer requires that the repository **must** be public (since no download authorization mechanism is implemented)

Reply:
If there will be necessity to support private repositories, we can ""easily"" (we have to actually sandbox each user own rules to avoid users to access not owned rules) add the user/org permission check